pkg/snapshot/snapshotter/snapshotter_test.go

```diff
@@ -570,192 +543,284 @@ var _ = Describe("Snapshotter", func() {
-			store, err = snapstore.GetSnapstore(snapstoreConfig)
-			Expect(err).ShouldNot(HaveOccurred())
-		})
-		Context("No previous snapshot was taken", func() {
-			It("should return true", func() {
-				snapshotterConfig := &brtypes.SnapshotterConfig{
-					FullSnapshotSchedule: fmt.Sprintf("%d %d * * *", (currentMin+1)%60, (currentHour+2)%24),
-				}
+			Describe("###GarbageCollectDeltaSnapshots", func() {
+				const (
+					deltaSnapshotCount = 6
+					testDir            = "garbagecollector_deltasnapshots.bkp"
+				)
 
-				ssr, err = NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
-				Expect(err).ShouldNot(HaveOccurred())
+				var snapshotterConfig *brtypes.SnapshotterConfig
 
-				// No previous snapshot was taken
-				ssr.PrevFullSnapshot = nil
-				isFullSnapMissed := ssr.IsFullSnapshotRequiredAtStartup(fullSnapshotTimeWindow)
-				Expect(isFullSnapMissed).Should(BeTrue())
-			})
-		})
-		Context("If previous snapshot was final full snapshot", func() {
-			It("should return true", func() {
-				snapshotterConfig := &brtypes.SnapshotterConfig{
-					FullSnapshotSchedule: fmt.Sprintf("%d %d * * *", (currentMin+1)%60, (currentHour+2)%24),
-				}
+				BeforeEach(func() {
+					snapshotterConfig = &brtypes.SnapshotterConfig{
+						FullSnapshotSchedule:     schedule,
+						DeltaSnapshotPeriod:      wrappers.Duration{Duration: 10 * time.Second},
+						DeltaSnapshotMemoryLimit: brtypes.DefaultDeltaSnapMemoryLimit,
+						GarbageCollectionPeriod:  wrappers.Duration{Duration: garbageCollectionPeriod},
+						GarbageCollectionPolicy:  brtypes.GarbageCollectionPolicyLimitBased,
+						MaxBackups:               maxBackups,
+					}
+				})
 
-				ssr, err = NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
-				Expect(err).ShouldNot(HaveOccurred())
+				AfterEach(func() {
+					err = os.RemoveAll(path.Join(outputDir, testDir))
+					Expect(err).ShouldNot(HaveOccurred())
+				})
 
-				// If previous snapshot was final full snapshot
-				ssr.PrevFullSnapshot = &brtypes.Snapshot{
-					IsFinal: true,
-				}
-				isFullSnapMissed := ssr.IsFullSnapshotRequiredAtStartup(fullSnapshotTimeWindow)
-				Expect(isFullSnapMissed).Should(BeTrue())
+				Context("with all delta snapshots older than retention period", func() {
+					It("should delete all delta snapshots", func() {
+						store := prepareStoreWithDeltaSnapshots(testDir, deltaSnapshotCount)
+						list, err := store.List()
+						Expect(err).ShouldNot(HaveOccurred())
+						Expect(len(list)).Should(Equal(deltaSnapshotCount))
+
+						ssr, err := NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
+						Expect(err).ShouldNot(HaveOccurred())
+
+						deleted, err := ssr.GarbageCollectDeltaSnapshots(list)
+						Expect(err).NotTo(HaveOccurred())
+						Expect(deleted).To(Equal(6))
+
+					})
```

Comment:
```suggestion
					It("should delete all delta snapshots", func() {
						store := prepareStoreWithDeltaSnapshots(testDir, deltaSnapshotCount)
						list, err := store.List()
						Expect(err).ShouldNot(HaveOccurred())
						Expect(len(list)).Should(Equal(deltaSnapshotCount))

						ssr, err := NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
						Expect(err).ShouldNot(HaveOccurred())

						deleted, err := ssr.GarbageCollectDeltaSnapshots(list)
						Expect(err).NotTo(HaveOccurred())
						Expect(deleted).To(Equal(6))
						list, err = store.List()
						Expect(err).ShouldNot(HaveOccurred())
						Expect(len(list)).Should(BeZero())
					})
```

At the end of each test case that you added, can you please list snapshots from the store again and check it's length? Just as another safety check to ensure the function is correct